src/one_pair_of_clients_version/client.py

```diff
@@ -126,6 +127,13 @@ def _get_other_player_name(self, player):
         return other_player
 
     def simulate_loading_with_spinner(self, loading_msg, loaded_json):
+
+        # In case this thread was started somewhere and is currently running, i.e there are two running threads with 
+        # this function as their target arguments, wait for the other one to complete before starting this one
+
+        if not self.spinner_thread_complete.is_set():
+            self.spinner_thread_complete.wait() 
+
```

Comment:
the simulate_loading_with_spinner thread could have multiple instances running at once, this makes sure that subsequent instance waits for preceding instance to finish before continuing